leshan-server-core/src/main/java/org/eclipse/leshan/server/bootstrap/BootstrapSessionManager.java

```diff
@@ -42,6 +43,29 @@
      */
     public BootstrapSession begin(String endpoint, Identity clientIdentity);
 
+    /**
+     * Called when we receive a successful response to a request.
+     * 
+     * @param request The request for which we get a successfull response.
+     */
+    public void onSuccess(LwM2mRequest<? extends LwM2mResponse> request);
```

Comment:
I think the method names can be confusing. e.g. onSuccess() is not about the success of the bootstrap sequence. Or onFailure() and the exisiting failed() are pretty close. 
Maybe onResponseSuccess(), onResponseError(), onRequestFailure()?

Reply:
:+1: